library/rsa.c

```diff
@@ -510,10 +510,13 @@ int mbedtls_rsa_gen_key( mbedtls_rsa_context *ctx,
 
     mbedtls_mpi_init( &H );
     mbedtls_mpi_init( &G );
+    mbedtls_mpi_init( &L );
 
     /*
      * find primes P and Q with Q < P so that:
      * GCD( E, (P-1)*(Q-1) ) == 1
+     *    and
+     * E^-1 mod LCM(P-1, Q-1) > N^0.5
```

Comment:
Thanks for adding precise references to FIPS 186-4 below, but I think this comment is still incorrect: `bitlen(d) >= nbits / 2` and `d > n^0.5` are not equivalent. Also, the block actually enforces 3 conditions and this comment only mentions 2, which I think is confusing. Please change to:
```
    /*
     * find primes P and Q with Q < P so that:
     *  abs(p-q) > 2^( nbits / 2 - 100 )
     *  GCD( E, (P-1)*(Q-1) ) == 1
     *  E^-1 mod LCM(P-1, Q-1) > N^0.5
     */
```